src/main/java/randoop/generation/ForwardGenerator.java

```diff
@@ -275,7 +281,17 @@ private ExecutableSequence createNewUniqueSequence() {
     }
 
     // Select a StatementInfo
-    TypedOperation operation = Randomness.randomMember(this.operations);
+    TypedOperation operation;
+
+    // If bloodhound is enabled, choose the next operation while considering the methods' weights.
+    if (GenInputsAbstract.enable_bloodhound) {
```

Comment:
I suggest generalizing this.
Create an interface with method `getNextOperation`.
Bloodhound will implement the interface, and so will a trivial class (named, say, `UniformRandomMethodSelection`) that just calls `Randomness.randomMember(this.operations);`.

In this class, set a variable of the interface type, to be either an instance of Bloodhound or of UniformRandom (depending on `GenInputsAbstract.enable_bloodhound`).  Then, here there doesn't have to be any special-casing for bloodhound.